ssl/ssl_sess.c

```diff
@@ -607,6 +609,7 @@ int ssl_get_prev_session(SSL_CONNECTION *s, CLIENTHELLO_MSG *hello)
                                         hello->pre_proc_exts, NULL, 0))
             return -1;
 
+        /* If we resumed, s->session will now be set */
```

Comment:
The new session comes from `tls_parse_extension(s, TLSEXT_IDX_psk, ...)`, so we're assuming under TLSv1.3 that we don't want any existing session, and will only use PSKs from the client?

Reply:
Right. It really doesn't make sense to set an existing session on a server SSL object. What would that mean??